.github/workflows/release.yml

```diff
@@ -1,63 +1,50 @@
             - main
     pull_request: # Runs on pull requests to any branch
 env:
-    IS_RELEASE_CANDIDATE: ${{ contains(github.head_ref, 'release--') }}
-    IS_FEATURE: ${{ !contains(github.head_ref, 'release--') }}
     DFX_VERSION: 0.21.0
     NODE_VERSION: 20
 jobs:
-    get-tests:
+    determine-should-release:
         if: ${{ contains(github.head_ref, 'release--') }}
+        name: Determine if this branch should release
+        runs-on: ubuntu-latest
+        outputs:
+            should_release: ${{ steps.determine_should_release.outputs.should_release }}
+        steps:
+            - uses: actions/checkout@v4
+
+            - id: determine_should_release
+              uses: ./.github/actions/should_release
+
+    get-tests:
+        needs: determine-should-release
+        if: ${{ contains(github.head_ref, 'release--') && needs.determine-should-release.outputs.should_release }}
         name: Get tests
         runs-on: ubuntu-latest
         outputs:
             tests: ${{ steps.get_tests.outputs.tests }}
         steps:
             - uses: actions/checkout@v4
+
             - name: Get all test dirs
               id: get_tests
-              uses: ./.github/actions/generate_tests
+              uses: ./.github/actions/get_tests
               with:
                   directories: |
                       ./examples
                       ./tests
 
-    determine-is-manual-release:
-        if: ${{ contains(github.head_ref, 'release--') }}
-        name: Determine if this job is for a manual release
-        runs-on: ubuntu-latest
-        outputs:
-            is_manual_release: ${{ steps.determine_is_manual_release.outputs.is_manual_release }}
-        steps:
-            - uses: actions/checkout@v4
-            - id: determine_is_manual_release
-              run: |
-                  BRANCH_NAME="${{ github.head_ref }}"
-                  RELEASE_VERSION="${BRANCH_NAME:9}"
-                  COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")
-
-                  IS_AUTOMATED_RELEASE="false"
-                  if [[ "$COMMIT_MESSAGE" == "azle-bot automated release $RELEASE_VERSION" ]]; then
-                      IS_AUTOMATED_RELEASE="true"
-                  fi
-
-                  if [[ "$IS_AUTOMATED_RELEASE" == "false" ]]; then
-                      echo "is_manual_release=false" >> "$GITHUB_OUTPUT"
-                  else
-                      echo "is_manual_release=true" >> "$GITHUB_OUTPUT"
-                  fi
-
-    release-candidate-deploy:
-        name: Deploy release candidate for release branches
+    release:
+        name: Deploy release
         # Only run this job if it's a release branch. This job will run instead of run-tests and will automatically publish another commit which will be tested
-        if: ${{ !needs.determine-is-manual-release.outputs.is_manual_release }}
+        if: ${{ needs.determine-should-release.outputs.should_release == 'true' }}
         needs:
             - get-tests
```

Comment:
Can we flip the order of these? Seems nice to put them in the order they will run...

Reply:
sure